src/aics-image-viewer/shared/utils/viewerState.ts

```diff
@@ -0,0 +1,60 @@
+import { Vector3 } from "three/src/Three";
+
+import {
+  ChannelSettingUpdater,
+  ChannelState,
+  ViewerSettingUpdater,
+  ViewerState,
+} from "../../components/ViewerStateProvider/types";
+
+export function overrideViewerState(changeViewerSetting: ViewerSettingUpdater, newState: ViewerState): void {
+  for (const key of Object.keys(newState) as (keyof ViewerState)[]) {
+    changeViewerSetting(key, newState[key] as any);
+  }
+
+  // Pathtrace rendering is not allowed in 2D view mode.
+  // Since we're not guaranteed on the order of object keys, handle the edge case
+  // where the viewer was in 2D mode and we are trying to set it to pathtrace + 3D mode, but
+  // render mode was set before the view mode.
+  changeViewerSetting("renderMode", newState.renderMode);
+}
+
+export function overrideChannelState(
+  changeChannelSetting: ChannelSettingUpdater,
+  index: number,
+  newState: ChannelState
+): void {
+  for (const key of Object.keys(newState) as (keyof ChannelState)[]) {
+    // Skip resetting name to default, since this causes channels to be dropped from the
+    // UI altogether.
+    if (key === "name") {
+      continue;
+    }
+    changeChannelSetting(index, key, newState[key] as any);
+  }
+}
+
+/** Returns the indices of any channels that have either the volume or isosurface enabled. */
+export function getEnabledChannelIndices(channelSettings: ChannelState[]): number[] {
+  const enabledChannels = [];
+  for (let i = 0; i < channelSettings.length; i++) {
+    if (channelSettings[i].volumeEnabled || channelSettings[i].isosurfaceEnabled) {
+      enabledChannels.push(i);
+    }
+  }
+  return enabledChannels;
+}
+
+/**
+ * Returns whether two subregions match in the X and Y axes.
+ */
+export function matchesSavedSubregion(subregion: Vector3 | null, savedSubregion: Vector3 | null): boolean {
+  // Fixes a bug where the initial saved subregion has a z=1 slice, but the the actual volume has slices chunked
+  // in z=2+ slices.
+  return (
+    savedSubregion !== null &&
+    subregion !== null &&
+    subregion.x === savedSubregion.x &&
+    subregion.y === savedSubregion.y
+  );
```

Comment:
In `App/index.tsx`, we are now applying a `requiredLoadSpec` on the initial volume load. In 2D mode, we specifically set the subregion to the Z-slice we are interested in. This causes the initial loaded slice to always have a Z-thickness of 1, and we save the subregion size as (x, y, 1).

HOWEVER, in volumes where Z-slices are chunked as 2 or more, this causes the reset behaviors to fail. There's an operation that waits until the channels for the saved subregion are loaded in, but it does a comparison between (x, y, 2) and (x, y, 1). To fix this, we skip comparing the z-dimension here.

Reply:
Maybe the above PR comment could be combined with the one in the code to augment it

Reply:
Noted, thanks!